inst/pages/98_exercises.qmd

```diff
@@ -1158,62 +1155,128 @@ See the OMA book for examples.
 
 ```{r}
+#| label: heatmap-visualization-ex2
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
 # Load necessary libraries
 library(SummarizedExperiment)
-library(pheatmap)
+library(sechm)
+library(compositions)
 
-# Example data creation
-# Assuming enterotype is a SummarizedExperiment object with sample metadata
-# and assay_data is a matrix containing assay data
+# Generate a sample SummarizedExperiment object 
+data(GlobalPatterns)
+tse <- GlobalPatterns
+
+# CLR transformation of abundance data matrix
+clr_data <- as.matrix(transformAssay(tse, method = "clr", assay.type = "counts", pseudocount = TRUE))
+
+# Hierarchical clustering on samples
+clustering <- hclust(dist(clr_data, method = "euclidean"), method = "average")
+clusters <- cutree(clustering, k = 3)
+colData(tse)$cluster <- as.factor(clusters)
+
+# Create annotation for clusters
+annotation_df <- data.frame(Cluster = colData(tse)$cluster)
+annotation_colors <- list(Cluster = c("1" = "red", "2" = "blue", "3" = "green"))
+
+# Visualize the heatmap with sechm
+sechm(
+  mat = clr_data,
+  row_labels = FALSE,
+  column_labels = FALSE,
+  row_cluster = clustering,
+  column_cluster = clustering,
+  row_annotation = annotation_df,
+  row_annotation_colors = annotation_colors,
+  main = "Heatmap with Clustering")
+```
 
-# Create example SummarizedExperiment object
-example_data <- matrix(runif(200), nrow = 20, ncol = 10) # Example assay data
-col_data <- DataFrame(Sample = paste0("Sample", 1:10), Group = rep(c("A", "B"), each = 5)) # Example sample metadata
-enterotype <- SummarizedExperiment(assays = list(counts = example_data), colData = col_data)
 
-# Extract sample metadata
-sample_metadata <- colData(enterotype)
+### 3: Heatmap with Annotations
 
-# Extract assay data
-assay_data <- assay(enterotype, "counts")
+- Uses heatmaps to visualize high-dimensional data, such as abundance data.
+- Annotations add context to the heatmap, linking clustering patterns to metadata like sample groups or experimental conditions.
 
-# Plot heatmap with clustering and sample annotations
-pheatmap(assay_data, cluster_rows = TRUE, cluster_cols = TRUE, annotation_col = as.data.frame(sample_metadata))
-```
+```{r}
+#| label: heatmap-visualization-ex3
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
 
-Make sure to replace the example data with your actual data. 
+# Load necessary libraries
+library(mia)
+library(SummarizedExperiment)
+library(sechm)
 
+# Generate a sample SummarizedExperiment object
+data(GlobalPatterns)
+tse <- GlobalPatterns
+
+# Ensure cluster information is correctly added to colData
+# (Assuming the clusters have been calculated previously and added to colData)
+
+# Create a sample metadata annotation (e.g., SampleType and Cluster)
+sample_metadata <- data.frame(SampleType = colData(tse)$SampleType,
+                              Cluster = colData(tse)$cluster)
+rownames(sample_metadata) <- colnames(assay(tse, "counts"))
```

Comment:
Is this really needed when we can call `as.data.frame(colData(tse))` or perhaps even just `colData(tse)`